### ISSUE TITLE
Revisit autoupdate and versioning logic

### ISSUE THREAD

**Issue by mbastian** [gephi/gephi#2444]
### Problem

We currently rely on the `gh-pages` branch of `gephi` to serve the autoupdate feature. All the Gephi and Netbeans NBM files are present [there](https://github.com/gephi/gephi/tree/gh-pages/0.9.2/netbeans_site). This pose a certain number of issues:
- The repository is already large and growing. Each new release adds 100MBs or so of files.
- The Maven plugin we use to upload those NBMs is not 100% reliable with those large repos
- We don't have an automated process to support hotfixes of releases

In addition, @wetneb remarked that we don't properly follow the Maven guidelines in terms of versioning. The 0.9.3 release for instance should rather have the minorVersion incremented. It should rather be `0.10.0`.

#### Versioning
* Gephi versions, in terms of what is communicated to users should be limited to the major + minor version. The patch version should be incremented for hotfixes and small (urgent) bugfixes that we want to still deploy to users.
* For instance, next version of Gephi should be known as the "0.10" version. The real, Maven version should be `0.10.0`. If a patch is made, it should then be `0.10.1` etc.
* Users shouldn't need to re-install Gephi for each patch release. This is exactly what we have the autoupdate for. The `0.10.0` version of Gephi should autoupdate to `0.10.1` etc. User preferences should also persist. In other words, the user data folder should be `0.10` and not `0.10.0`.

#### Autoupdate
* Module NBMs are already uploaded to Maven central as part of the release process. Ideally, the autoupdate `updates.xml` file points at those files instead of having to upload them separately again.
* A separate `gephi-repository` repo is used, to not overload the `gephi` repo.

**Comment by mbastian** [comment]
Update on this front.

**Implemented:**
* The `updates.xml` directly points at the Maven central URL instead of the same folder. This is a major win! We no longer need to upload the NBMs to our own repository. Only the `updates.xml` need to be refreshed - the NBMs are already on Maven central as part of the release process.
* The Update Center URL in Gephi has been updated to only contain Gephi's minor version. For instance `https://raw.githubusercontent.com/gephi/gephi/gh-pages/0.9/autoupdate/updates.xml`. This means future _patch_ versions like 0.9.4, 0.9.5, etc. will use the same Update Center source. This in turn will allow 0.9.3 users to autoupdate to 0.9.4 without having to re-install Gephi. Gephi will automatically inform them of the new version, download the NBMs, reboot and done. This is also a win for users and will also make it easier for us to release more regularly.
* User preferences folders now also contain Gephi's minor version. This means user preferences will be kept between versions 0.9.3, 0.9.4 etc. A 0.10.0. version would require a re-installation of Gephi, but the `SettingsUpgrader` module would take care of transferring the preferences. No changes there.
* As part of the release GH actions, we now use the [Maven SCM Plugin](https://maven.apache.org/scm/maven-scm-plugin/) instead of the [Git Site Plugin](https://github.github.com/maven-plugins/site-plugin/) to push the `updates.xml` file. A lot more reliable.
* Autoupdate files are generated also for development versions (i.e. when containing -SNAPSHOT in version). That was a nice to have, but allows us to test things end-to-end ahead of releasing.

**Not implemented:**
* We sticked to the `gh-pages` branch of this repository as opposed to creating a separate `gephi-repository` Git repository. The main reason is because we no longer need to store the NBMs so the repository size should no longer be an issue. The other reason is because of GitHub actions. It's significantly harder to upload files to another Git repository than the one you're building from. I ran into permissions issues.

**To be noted:**
* Previously when we did hotfixes after a release we incremented the version number of the modules we changed manually, and released those NBMs to the autoupdate repository. This was somewhat manual and error prone but it had the advantage of ONLY updating the NBMs that actually changed. With this new process, when a new patch version is released (e.g. 0.9.4) ALL modules will be updated to this new version. A bit more heavy for users to download all the modules.
* It may be a bit confusing for future 0.9 patch versions as we've changed the versioning concept between 0.9.2 and 0.9.3. Not ideal, but everything should become clear once we release the 0.10.0 version. From that point on, we can remain consistent for the years to come.
